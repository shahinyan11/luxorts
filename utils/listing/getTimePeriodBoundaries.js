import moment from 'moment';

import { DATE_FORMAT } from 'constants';
import isPresent from 'utils/isPresent';

const getTimePeriodBoundaries = (date) => {
  const baseDate = isPresent(date) ? moment(date) : moment();

  return {
    startDate: baseDate.clone().subtract(1, 'months').startOf('months').format(DATE_FORMAT),
    endDate: baseDate.clone().add(1, 'months').endOf('months').format(DATE_FORMAT),
  };
};

export default getTimePeriodBoundaries;
