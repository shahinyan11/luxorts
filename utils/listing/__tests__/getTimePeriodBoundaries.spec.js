import getTimePeriodBoundaries from '../getTimePeriodBoundaries';

it('getTimePeriodBoundaries util matches snapshot', () => {
  expect(getTimePeriodBoundaries(new Date('15 Jun 2022 00:00:00 GMT'))).toMatchSnapshot();
});
