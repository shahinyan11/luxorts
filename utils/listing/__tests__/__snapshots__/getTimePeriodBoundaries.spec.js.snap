// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTimePeriodBoundaries util matches snapshot 1`] = `
Object {
  "endDate": "31/07/2022",
  "startDate": "01/05/2022",
}
`;
